data/account/participation.go

```diff
@@ -56,6 +57,24 @@ type Participation struct {
 	KeyDilution uint64
 }
 
+// ParticipationID computes a ParticipationID.
+func (part Participation) ParticipationID() ParticipationID {
+	data := new(bytes.Buffer)
+
+	data.Write(part.Parent[:])
+	binary.Write(data, binary.LittleEndian, part.FirstValid)
+	binary.Write(data, binary.LittleEndian, part.LastValid)
+	binary.Write(data, binary.LittleEndian, part.KeyDilution)
+	if part.VRF != nil {
+		data.Write(part.VRF.PK[:])
+	}
+
+	// this too?
+	//part.Write(part.Voting.SubKeyPK[:])
+
+	return ParticipationID(crypto.Hash(data.Bytes()))
+}
```

Comment:
The common way we calculate a hash of a structure is by msgp-ing it. Are you thinking of any particular reason we shouldn't do it here ?

Reply:
My understanding is that the subkeys will change over time, so this was an attempt to make sure the same keyset wouldn't produce different ParticipationIDs

Reply:
While what you just wrote is correct, it's not what I meant.
I was thinking of -
```golang
serializing := Participation{
   FirstValid: part.FirstValid,
   ...
```
while omitting the changing parts ( i.e. Voting keys ).

then
```golang
   return ParticipationID(crypto.Hash(serializing.Marshal(nil)))
```